lib/ParseSIL/ParseSIL.cpp

```diff
@@ -6742,6 +6745,242 @@ bool SILParserTUState::parseSILDefaultWitnessTable(Parser &P) {
+    // Parse parameter index list.
+    SmallVector<unsigned, 8> paramIndices;
+    // Function that parses an index into `paramIndices`. Returns true on error.
+    auto parseParam = [&]() -> bool {
+      unsigned index;
+      // TODO: Reject non-ascending index lists.
+      if (P.parseUnsignedInteger(index, lastLoc,
+              diag::sil_diff_witness_expected_index_list))
+        return true;
+      paramIndices.push_back(index);
+      return false;
+    };
+    // Parse first.
+    if (parseParam())
+      return true;
+    // Parse rest.
+    while (P.Tok.isNot(tok::r_square))
+      if (parseParam())
+        return true;
+    if (P.parseToken(tok::r_square, diag::sil_diff_witness_expected_keyword,
+                     "]"))
+      return true;
+    auto maxIndexRef =
+        std::max_element(paramIndices.begin(), paramIndices.end());
+    indexSubset = IndexSubset::get(
+        P.Context, maxIndexRef ? *maxIndexRef + 1 : 0, paramIndices);
+    return false;
+  };
+  // Parse parameter and result indices.
+  IndexSubset *parameterIndices = nullptr;
+  IndexSubset *resultIndices = nullptr;
+  if (parseIndexSubset("parameters", parameterIndices))
+    return true;
+  if (parseIndexSubset("results", resultIndices))
+    return true;
+
+  // Parse a trailing 'where' clause (optional).
+  // This represents derivative generic signature requirements.
+  GenericSignature *derivativeGenSig = nullptr;
+  SourceLoc whereLoc;
+  SmallVector<RequirementRepr, 4> derivativeRequirementReprs;
+  if (P.Tok.is(tok::l_square) && P.peekToken().is(tok::kw_where)) {
+    P.consumeToken(tok::l_square);
+    bool firstTypeInComplete;
+    P.parseGenericWhereClause(whereLoc, derivativeRequirementReprs,
+                              firstTypeInComplete,
+                              /*AllowLayoutConstraints*/ false);
+    if (P.parseToken(tok::r_square, diag::sil_diff_witness_expected_keyword,
+                     "]"))
+      return true;
+  }
+
+  // Parse original function name.
+  SILFunction *originalFn;
+  if (parseFunctionName(originalFn))
+    return true;
+
+  // Resolve derivative requirements.
+  if (!derivativeRequirementReprs.empty()) {
+    SmallVector<Requirement, 4> requirements;
+    auto *whereClause = TrailingWhereClause::create(
+        originalFn->getModule().getASTContext(), whereLoc,
+        derivativeRequirementReprs);
+    convertRequirements(P, originalFn, whereClause->getRequirements(),
+                        requirements);
+    assert(requirements.size() == derivativeRequirementReprs.size());
+    derivativeGenSig = evaluateOrDefault(
+        P.Context.evaluator,
+        AbstractGenericSignatureRequest{
+            originalFn->getLoweredFunctionType()->getGenericSignature(),
+            /*addedGenericParams=*/{},
+            std::move(requirements)},
+            nullptr);
+  }
+
+  // Parse differentiability witness body.
+  SILFunction *jvp = nullptr;
+  SILFunction *vjp = nullptr;
+  if (P.Tok.is(tok::l_brace)) {
```

Comment:
```suggestion
  SourceLoc lBraceLoc;
  if (P.consumeIf(Tok::l_brace, lBraceLoc)) {
```

Reply:
Done in https://github.com/apple/swift/pull/27487/commits/2073f863826fb305c1fc105a520d84de1e1463ec.